ci/nostd-check/rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "nightly"
```

Comment:
Why `nightly` is needed here?

Reply:
By reading through, for the `#[global_allocator]` use in the test, but I'm also confused as to why override it

Reply:
I couldn't find any simple allocator that works in Rust stable. The alternative is define our own dummy one that does completely nothing (it just needs to compile, it doesn't need to actually run). By the way, as I did it the nightly toolchain is applied only to the `nostd-check` crate.

Reply:
@p-avital Just saw your message. The target `x86_64-unknown-none` does not have a default allocator, hence Rust complains if none is defined (since Zenoh uses the `alloc` crate)